### PR TITLE
add CORS support

### PR DESCRIPTION
**Pull request by kevbob1**
I had a need to hit a zoorest instance from an Angular browser app.  I included a module that implements the CORS protocol.    I am open to ideas on how to improve this feature.  (make it optional via cli switch?)

### CODE REVIEW

File: rest/api.go [hunk 1]
```diff
@@ -226,10 +227,19 @@ func Serve(listen string, zk ZooNode) {
 	r.HandleFunc("/v2{path:[A-Za-z0-9-_/.:]+}", zk.RM).Methods("DELETE")
 	r.HandleFunc("/v2{path:[A-Za-z0-9-_/.:]+}", zk.UP).Methods("PUT", "POST", "PATCH")
 
-	http.Handle("/", r)
+	c := cors.New(cors.Options{
+		AllowedMethods: []string{"GET", "LIST", "DELETE", "PUT", "POST", "PATCH"},
+
+		// Enable Debugging for testing, consider disabling in production
+		Debug: true,
```
Comment (Difrex) [inline rest/api.go]: Make debug flag please for turn on/off a debug mode

File: rest/api.go [hunk 2]
```diff
@@ -226,10 +227,19 @@ func Serve(listen string, zk ZooNode) {
 	r.HandleFunc("/v2{path:[A-Za-z0-9-_/.:]+}", zk.RM).Methods("DELETE")
 	r.HandleFunc("/v2{path:[A-Za-z0-9-_/.:]+}", zk.UP).Methods("PUT", "POST", "PATCH")
 
-	http.Handle("/", r)
+	c := cors.New(cors.Options{
```
Comment (Difrex) [inline rest/api.go]: This is for the same domain or for a '*'?

Maybe we can make it more configurable?

**Review by Difrex — COMMENTED** [review]
Thank you for your PR!

There are some questions about  "github.com/rs/cors".

This package can accept more options instead of `Debug`.
Maybe we can configure it? What do you think about environment variables?

smth like this:
```go
const (
  CorsDebugModeEnvVar string = `ZOOREST_CORS_DEBUG_ENABLE`
  // CorsAnotherOptEnvVar ...
)

corsOpt := os.Getenv(CorsDebugModeEnvVar)
if strings.toLower(corsOpt) == strings.toLower("true") {
  // enable cors option
} 
```

Or we can make other flags to enabling/disabling cors options.

**Comment by kevbob1** [comment]
+1 for environment variables.  This makes it easily configurable for docker/container deployments.  

**ZOOREST_CORS_ENABLE**    (enable the feature.  Any non-null value is considered "true" and enables it.    **default:  false**)
**ZOOREST_CORS_DEBUG_ENABLE**  (enable CORS debug mode Any non-null value is considered "true" and enables it.   **default: false**)
**ZOOREST_CORS_ALLOWED_ORIGINS** (comma delimited list of origin url patterns to allow access to the service.  **default: \***)

**Comment by Difrex** [comment]
Sounds good.

Can you make these changes, please?

**Comment by kevbob1** [comment]
Hi, yes i will make the changes and update the PR.   Thanks!
---
Kevin

**Pull request merged**